src/db/serde.ts

```diff
@@ -8,25 +16,51 @@ export interface Hook<U, V> {
 const DESERIALIZE_HOOK: Hook<any, any> = {};
 const SERIALIZE_HOOK: Hook<any, any> = {};
 
-function applyHook<T extends Table, U, V>(
-  hook: Hook<U, T>,
-  table: Table,
-  values: InsertableForTable<T> | InsertableForTable<T>[]
-): InsertableForTable<T> | InsertableForTable<T>[] {
-  return Array.isArray(values)
-    ? values.map((v) => applyHookSingle(hook, table, v))
-    : applyHookSingle(hook, table, values);
+type InsertableOrSelectableForTable<T extends Table> =
+  | InsertableForTable<T>
+  | SelectableForTable<T>;
+type InsertableOrSelectableForTableArray<T extends Table> =
+  | InsertableForTable<T>[]
+  | SelectableForTable<T>[];
+
+function applyHook<
+  T extends Table,
+  V extends
+    | InsertableOrSelectableForTable<T>
+    | InsertableOrSelectableForTableArray<T>,
+  U,
```

Comment:
are these types actually important? nothing else depends on U or W...I think hook can just be `hook: Hook<unknown, unknown>` or `hook: Hook<any, any>` if unknown is too restrictive

Reply:
Yeah, this is a good point. It's optimistic re: being able to type it. Filed #8.